src/modules/launcher/Plugins/Community.PowerToys.Run.Plugin.IPLookup/Main.cs

```diff
@@ -0,0 +1,117 @@
+﻿// Copyright (c) Microsoft Corporation
+// The Microsoft Corporation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.InteropServices;
+using System.Windows;
+using System.Windows.Controls;
+using ManagedCommon;
+using Wox.Plugin;
+using Wox.Plugin.Logger;
+
+namespace Community.PowerToys.Run.Plugin.IPLookup
+{
+    public class Main : IPlugin
+    {
+        private PluginInitContext? _context;
+        private string? _iconPath;
+
+        public string Name => Properties.Resources.iplookup_plugin_name;
+
+        public string Description => Properties.Resources.iplookup_plugin_description;
+
+        public void Init(PluginInitContext context)
+        {
+            _context = context ?? throw new ArgumentNullException(nameof(context));
+            UpdateIconPath(_context.API.GetCurrentTheme());
+        }
+
+        private void UpdateIconPath(Theme theme)
+        {
+            if (theme == Theme.Light || theme == Theme.HighContrastWhite)
+            {
+                _iconPath = "Images/IPLookup_icon_light.png";
+            }
+            else
+            {
+                _iconPath = "Images/IPLookup_icon_dark.png";
+            }
+        }
+
+        public List<Result> Query(Query query)
```

Comment:
We should use delayed query: `        public List<Result> Query(Query query, true)`